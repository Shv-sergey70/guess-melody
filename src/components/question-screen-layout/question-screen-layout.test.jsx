import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import QuestionScreenLayout from "./question-screen-layout";

configure({adapter: new Adapter()});

describe(`Game screen correctly works`, () => {
  const genreQuestion = {
    type: `genre`,
    genre: `rock`,
    answers: [
      {
        src: `./tracks/Anivar_Summer.mp3`,
        genre: `pop`
      },
      {
        src: `./tracks/Billie_Eilish_ilomilo.mp3`,
        genre: `jazz`
      },
      {
        src: `./tracks/Cj_L_Black_Rain.mp3`,
        genre: `rock`
      },
      {
        src: `./tracks/Orquesta_L_rica_de_Barcelona_Las_Cuatro_Estaciones,_Invierno_Allegro_Non_Molto.mp3`,
        genre: `classic`
      }
    ]
  };

  const artistQuestion = {
    type: `artist`,
    song: {},
    answers: [
      {
        picture: `/img/default-artist.png`,
        artist: `Billie Eilish`
      },
      {
        picture: `/img/default-artist.png`,
        artist: `Imagine Dragons`
      },
      {
        picture: `/img/default-artist.png`,
        artist: `Artur Latte`
      }
    ]
  };

  test(`Genre screen renders`, () => {
    const tree = shallow(
        <QuestionScreenLayout
          question={genreQuestion}
          onAnswer={jest.fn()}
        />
    );

    expect(tree).toMatchSnapshot();
  });

  test(`Artist screen renders`, () => {
    const tree = shallow(
        <QuestionScreenLayout
          question={artistQuestion}
          onAnswer={jest.fn()}
        />
    );

    expect(tree).toMatchSnapshot();
  });
});
