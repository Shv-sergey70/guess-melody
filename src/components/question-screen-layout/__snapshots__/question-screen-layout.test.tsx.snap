// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game screen correctly works Artist screen renders 1`] = `ShallowWrapper {}`;

exports[`Game screen correctly works Genre screen renders 1`] = `ShallowWrapper {}`;
